source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,255 @@
+maximum number of connections the driver can maintain in the connection pool.
+
+.. example::
+
+   Consider a failover scenario in a 3 member replica set with applications under
+   heavy load. Without ``maxConnecting``, ``maxPoolSize * application servers``
+   connections would be established. With default settings for ``maxPoolSize``
+   and 3 application servers, that would be 300 new connections, concurrently.
+
+   With ``maxConnecting`` set to 3, the new primary will receive
+   ``maxConnecting * application servers`` connections concurrently, or 9.
+
+Specify ``maxConnecting`` in your connection URI or with the
+`ConnectionPoolSettings.Builder maxConnecting() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html#maxConnecting(int)>`__ method.
+
+maxPoolSize
++++++++++++
+
+:ref:`Properly sizing your connection pool <connection-pool-sizing-java>`
+helps you avoid the risk of connection storms by limiting the amount of
+connections any application can establish.
+
+Specify ``maxPoolSize`` in your connection URI or with the
+`ConnectionPoolSettings.Builder maxSize() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html#maxSize(int)>`__ method.
+
+maxTime()
++++++++++
+
+.. tip::
+
+   Use ``maxTime()`` to limit operation execution time. Do not use ``socketTimeoutMS``.
+
+``maxTime()`` specifies the *cumulative* time limit for processing operations
+in MongoDB. If an operation exceeds this limit, MongoDB terminates the
+operation.
+
+``maxTime()`` is available on the following classes:
+
+- `AggregateIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/AggregateIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `DistinctIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/DistinctIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `FindIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/FindIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `ListCollectionsIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/ListCollectionsIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `ListDatabasesIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/ListDatabasesIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `ListIndexesIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/ListIndexesIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `MapReduceIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MapReduceIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+- `GridFSFindIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/gridfs/GridFSFindIterable.html#maxTime(long,java.util.concurrent.TimeUnit)>`__
+
+.. warning:: Do not use ``socketTimeoutMS`` to abort operations.
+
+   Some users have incorrectly used this option to try to abort operations
```

Comment:
I wonder if this information should be incorporated in the "tip" admonition earlier in the section since it already suggests not using "socketTimeoutMS" to limit operation execution time.

Also, I think "some users" is unspecific (similar to "many people say..."), and pointing out their incorrect usage indicates some blame, which isn't useful.